frontend/packages/console-shared/src/hooks/scroll.ts

```diff
@@ -0,0 +1,53 @@
+import * as React from 'react';
+
+export enum ScrollDirection {
+  scrollingUp = 'scrolling-up',
+  scrollingDown = 'scrolling-down',
+  scrolledToBottom = 'scrolled-to-bottom',
+  scrolledToTop = 'scrolled-to-top',
+}
+
+export const useScrollDirection = (node: HTMLElement) => {
+  const scrollPosition = React.useRef(null);
+  const [scrollDirection, setScrollDirection] = React.useState<ScrollDirection>(null);
+  const handleScroll = React.useCallback(
+    (event) => {
+      const { scrollHeight, scrollTop, clientHeight } = event.target;
+      if (!scrollPosition.current) {
+        scrollPosition.current = scrollTop;
+        return;
+      }
+      if (scrollHeight - scrollTop === clientHeight) {
+        scrollPosition.current = scrollTop;
+        setScrollDirection(ScrollDirection.scrolledToBottom);
+      }
+      if (scrollTop === 0) {
+        scrollPosition.current = scrollTop;
+        setScrollDirection(ScrollDirection.scrolledToTop);
+      }
+      if (scrollPosition.current > scrollTop) {
+        scrollPosition.current = scrollTop;
+        if (scrollDirection !== ScrollDirection.scrollingUp)
+          setScrollDirection(ScrollDirection.scrollingUp);
+      }
+      if (scrollPosition.current < scrollTop) {
+        scrollPosition.current = scrollTop;
+        if (scrollDirection !== ScrollDirection.scrollingDown)
+          setScrollDirection(ScrollDirection.scrollingDown);
+      }
+    },
+    [scrollDirection],
+  );
+
+  React.useEffect(() => {
+    if (node) {
+      node.addEventListener('scroll', (event) => setTimeout(() => handleScroll(event), 100));
+    }
+    return () => {
+      clearTimeout();
+      node && node.removeEventListener('scroll', handleScroll);
```

Comment:
This isn't removing the `eventListener`... needs to be the same function ref... Your function is the lambda to a setTimeout.

Reply:
Done